### ISSUE TITLE
"reference not set to an instance of an object." when disconnecting from server

### ISSUE THREAD

**Issue by gbakeman** [icechat/IceChat#8]
I'm experiencing an issue after updating to 9.08 where when I try to disconnect from a server, it gives the aforementioned error. The full message:

`[03:01.01] Error:ParseOutGoingCommand:Console:0:Depression is merely anger without enthusiasm:Object reference not set to an instance of an object.:   at IceChat.IRCConnection.SendData(String data)
    at IceChat.FormMain.SendData(IRCConnection connection, String data)
    at IceChat.FormMain.ParseOutGoing Command(IRCConnection connection, String data)
`
When that happened, I modified the server settings and replaced the quit message property with a space character ' ', which now gives the following errors:
`
[03:01.36] Error:ParseOutGoingCommand:Console:0: :Objectreference not set to an instance of an object.:   at IceChat.IRCConnection.SendData(String data)
    at IceChat.FormMain.SendData(IRCConnection connection, String data)
    at IceChat.FormMain.ParseOutGoingCommand(IRCConnection connection, String  data)
`
Note that when I changed the server settings, I had not yet disconnected.

**Comment by icechat** [comment]
Does this happen every time you disconnect from a server, or occassionaly?

**Comment by gbakeman** [comment]
Only occasionally, unfortunately... I was using IceChat some more, and I couldn't actually reproduce it :(

**Comment by icechat** [comment]
Ok, thank you. How were you disconnecting from the server, using the /quit command, or the server tree?

**Comment by gbakeman** [comment]
I was using either the disconnect button on the server tree, or the x on the server tab. Both would seem to have the same result.

**Comment by icechat** [comment]
Ok, will see if I can duplicate error.

**Comment by icechat** [comment]
Did you have SSL enabled for this server?  Just trying to narrow it down.

**Comment by gbakeman** [comment]
I recall attempting to enable it, but not having much success since I didn't know what the port was. I would have had it disabled when I was connected.

**Comment by icechat** [comment]
Thanks, I will just add more error reporting to that function, if it happens again.

**Comment by gbakeman** [comment]
@icechat I was just using IceChat today, and I think I've managed to reliably reproduce this issue.

Essentially, I connected to a server. Then I started editing the server properties. When I attempt to disconnect, it says:
`6:11:18 PMError:SendData(s) Error:QUIT :Light travels fasterthen sound, which is why some people appear bright, until you hear them speak:Object reference not set to an instance of an object.:   at IceChat.IRCConnection.SendData(String data)`
I modified the server address, port, and SSL/IPv6 tick boxes. Perhaps Changing the address is causing this?
